naga/src/lib.rs

```diff
@@ -1900,6 +1919,15 @@ pub enum Statement {
         /// [`AtomicResult`]: crate::Expression::AtomicResult
         result: Handle<Expression>,
     },
+    /// Atomic function with no return value.
+    AtomicNoReturn {
+        /// Pointer to an atomic value.
+        pointer: Handle<Expression>,
+        /// Function to run on the atomic.
+        fun: AtomicFunctionNoReturn,
+        /// Value to use in the function.
+        value: Handle<Expression>,
+    },
```

Comment:
Let's get some proper documentation on this. What does it do? What types does it support? What address spaces are permitted under which features? Why is it different from `Atomic`?

I've made one attempt at reading through the code in `valid/function.rs`, but I'm still struggling to understand exactly which behaviors this statement supports and does not support. Ideally the documentation in `lib.rs` should fully explain the behavior and constraints of each statement.